src/services/users.js

```diff
@@ -0,0 +1,165 @@
+const { User } = require('../models');
+const { UserRoles } = require('../models');
+const { UserStatuses } = require('../models');
+const bcrypt = require('bcryptjs');
+
+class UsersService {
+  async updateUserPassword (req, res) {
+    try {
+      const { oldPassword, newPassword } = req.body;
+      const user = await User.findOne({ _id: req.params.id }).exec();
+
+      bcrypt.compare(oldPassword, user.password, async function (err, isMatch) {
+        if (err) throw err;
+
+        if (isMatch && oldPassword !== newPassword) {
+          user.password = newPassword;
+          const salt = await bcrypt.genSalt(10);
+          user.password = await bcrypt.hash(user.password, salt);
+          await user.save();
+          return user ? res.json(true) : res.boom.notFound();
+        } else {
+          res.boom.badRequest();
+        }
+      });
+    } catch (err) {
+      res.boom.badRequest();
```

Comment:
Add some messages to an error. There must an errors like: "new password cannot be the same as old", "wrong old password"

Reply:
fixed